inst/include/RcppArmadilloAs.h

```diff
@@ -403,14 +403,68 @@ namespace traits {
                     res = symmatl(res);
                 }
             }
+            else if (type == "pMatrix") {
+                IntegerVector i;
+                IntegerVector p(ncol + 1);
+                IntegerVector x(ncol, 1);
+                IntegerVector perm = mat.slot("perm");
+                
+                // Sort the row number by the column number
+                std::map <int, int> colrow;
+                for(int tmp = 0; tmp < perm.size(); tmp++){
+                    colrow[perm[tmp]] = tmp;
+                }
+                
+                // Calculate i
+                for(std::map<int, int>::iterator tmp = colrow.begin(); tmp != colrow.end(); tmp++){
+                    i.push_back(tmp -> second);
+                }
+              
+                // Calculate p
+                for(int tmp = 0; tmp < p.size(); tmp++){
+                    p[tmp] = tmp;
+                }
+              
+                // Making space for the elements
+                res.mem_resize(static_cast<unsigned>(x.size()));
+              
+                // Copying elements
+                std::copy(i.begin(), i.end(), arma::access::rwp(res.row_indices));
+                std::copy(p.begin(), p.end(), arma::access::rwp(res.col_ptrs));
+                std::copy(x.begin(), x.end(), arma::access::rwp(res.values));
+            }
+            else if (type == "ddiMatrix") {
+                IntegerVector i(ncol);
```

Comment:
This also needs to be tweaked slightly as it does not take into account `N` or `U`  classes of the [`ddiMatrix`](https://stat.ethz.ch/R-manual/R-devel/library/Matrix/html/ddiMatrix-class.html).


e.g.

```r
(d_U <- Diagonal(n = 3))
# 3 x 3 diagonal matrix of class "ddiMatrix"
#     [,1] [,2] [,3]
# [1,]    1    .    .
# [2,]    .    1    .
# [3,]    .    .    1
str(d_U)
# Formal class 'ddiMatrix' [package "Matrix"] with 4 slots
#  ..@ diag    : chr "U"
#  ..@ Dim     : int [1:2] 3 3
#  ..@ Dimnames:List of 2
#  .. ..$ : NULL
#  .. ..$ : NULL
#  ..@ x       : num(0) 
```
```
(d_N <- Diagonal(x = c(10,1)))
# 2 x 2 diagonal matrix of class "ddiMatrix"
#      [,1] [,2]
# [1,]   10    .
# [2,]    .    1
str(d_N)
# Formal class 'ddiMatrix' [package "Matrix"] with 4 slots
#  ..@ diag    : chr "N"
#  ..@ Dim     : int [1:2] 2 2
#  ..@ Dimnames:List of 2
#  .. ..$ : NULL
#  .. ..$ : NULL
#  ..@ x       : num [1:2] 10 1
```